### DISCUSSION TITLE
Is there any docs about understansing source code?

### DISCUSSION THREAD

**Discussion by ssslambda** [chakra-ui/chakra-ui#2638] [Q&A]
I would to build something similar to Chakra (it's awesome tbh) for my project (it has design system), but, it's hard to understand Chakra source code, i know that this is not related to Chakra itself, but it would be very very nice (and i gonna love it) if it has description about internals in README (like i saw in nodejs.dev repo, but more detailed, Chakra is even complex), or other place which isn't hard to find (i looked at contribution guide and didn't find something similar).

**Comment by chr-ge** [comment]
Hey, there are no docs that really explain the source code but it's written in typescript which should make it easier to make sense of it. When you look at the files look at the interfaces and types and all the comments that the devs have left.

**Reply by ssslambda** [reply]
It's very hard to navigate between code base without any map, just (very big i think) README in packages folder would be really helpfull.

**Comment by with-heart** [comment]
Unfortunately we don't, but this has been on my todo list for a very long time now. We even have a GitHub project with a number of cards in it that may be relevant. https://github.com/chakra-ui/chakra-ui/projects/3

I have attached a decent chunk of notes to some of them that might help you get started in understanding how the system works. Someday this will hopefully become official documentation, or at least have a general overview to make it more digestible.

**Reply by with-heart** [reply]
Feel free to ask additional questions about the source here or [in our Discord](https://discord.com/invite/dQHfcWF) and I'll do my best to point you in the right direction.

**Comment by iamgabrielsoft** [comment]
is more like going to a journey without no direction... hahaha!
